cmd/virt-handler/virt-handler.go

```diff
@@ -305,6 +301,10 @@ func (app *virtHandlerApp) Run() {
 	app.clusterConfig.SetConfigModifiedCallback(app.shouldChangeLogVerbosity)
 	app.clusterConfig.SetConfigModifiedCallback(app.shouldChangeRateLimiter)
 
+	if err := app.setupTLS(factory); err != nil {
```

Comment:
Yes, or at least it depends on where it will be moved. Basically it is necessary that when you make this function call, the `factory` parameter has everything initialized.